src/llm_vm/utils/keys.py

```diff
@@ -45,7 +45,8 @@ def set_api_key(key, key_type="OPENAI_API_KEY"):
                 - GOOGLE_KEY
                 - GOOGLE_CX
     """
-    os.environ[key_type]=key
+    if not os.getenv(key_type):
```

Comment:
@abhigya-sodani @cartazio the bug was here, where we were setting the openai env variable while initializing the rebel agent. Because rebel was initialized w/out a key, the user's env. variable ended up getting blanked preventing calls to open ai.